libbpf-tools/trace_helpers.c

```diff
@@ -150,3 +150,70 @@ const struct ksym *ksyms__get_symbol(const struct ksyms *ksyms,
 
 	return NULL;
 }
+
+#define min(x, y) ({				 \
+	typeof(x) _min1 = (x);			 \
+	typeof(y) _min2 = (y);			 \
+	(void) (&_min1 == &_min2);		 \
+	_min1 < _min2 ? _min1 : _min2; })
+
+static void print_stars(unsigned int val, unsigned int val_max, int width)
+{
+	int num_stars, num_spaces, i;
+	bool need_plus;
+
+	num_stars = min(val, val_max) * width / val_max;
+	num_spaces = width - num_stars;
+	need_plus = val > val_max;
+
+	for (i = 0; i < num_stars; i++)
+		printf("*");
+	for (i = 0; i < num_spaces; i++)
+		printf(" ");
+	if (need_plus)
+		printf("+");
+}
+
+void print_log2_hist(unsigned int *vals, int vals_size, char *val_type)
+{
+	int stars_max = 40, idx_max = -1;
+	unsigned int val, val_max = 0;
+	long low, high;
+	int stars, i;
+
+	for (i = 0; i < vals_size; i++) {
+		val = vals[i];
+		if (val > 0)
+			idx_max = i;
+		if (val > val_max)
+			val_max = val;
+	}
+
+	if (idx_max > 0) {
+		if (idx_max <= 32)
+			printf("     %-19s : count    distribution\n",
+				val_type);
+		else
+			printf("               %-29s : count     "
+				"distribution\n", val_type);
+	}
+
+	if (idx_max <= 32)
+		stars = stars_max;
+	else
+		stars = stars_max / 2;
+
+	for (i = 1; i < idx_max + 1; i++) {
+		low = ((long)1 << i) >> 1;
+		high = ((long)1 << i) - 1;
```

Comment:
pro tip: `(long)1` is the same as `1L`. But it can be 32-bit on 32-bit arches. So I'd go with long long: `1LL`. And you probably want unsigned integers everywhere?

Reply:
Yes, thank you for pointing this.

Reply:
Fixed, thanks